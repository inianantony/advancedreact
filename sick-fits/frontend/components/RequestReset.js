import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage'
import Router from 'next/router'
import styled from 'styled-components'
import { CURRENT_USER_QUERY } from './User'

const REQUEST_RESET_MUTATION = gql`
    mutation REQUEST_RESET_MUTATION($email: String!) {
        requestReset(
            email: $email
        ){
            message
        }
    }
`;

export default class RequestReset extends Component {
    state = {
        email: '',
    }
    handleChange = (e) => {
        const { name, value } = e.target;
        this.setState({ [name]: value });
    }
    render() {
        return (
            <Mutation mutation={REQUEST_RESET_MUTATION} variables={this.state}>
                {(requestReset, { loading, error, called }) => (
                    <Form method="POST" onSubmit={async (e) => {
                        e.preventDefault();
                        const res = await requestReset();
                        this.setState({ email: '' });
                    }}>
                        <Error error={error} />
                        {!error && !loading && called && <p>Success! Check your email for reset link</p>}
                        <fieldset disabled={loading} aria-busy={loading}>
                            <h2>Sign In</h2>
                            <label htmlFor="email">
                                Email
                            <input type="email" id="email" name="email" placeholder="Email" required value={this.state.email} onChange={this.handleChange} />
                            </label>
                            <button type="Submit">Reset Password</button>
                        </fieldset>
                    </Form>
                )}
            </Mutation>
        );
    }
}