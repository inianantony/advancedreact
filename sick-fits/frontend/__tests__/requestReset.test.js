import { mount } from 'enzyme';
import wait from 'waait';
import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';
import { MockedProvider } from 'react-apollo/test-utils';
import toJSON from 'enzyme-to-json';

const mocks = [{
    request: { query: REQUEST_RESET_MUTATION, variables: { email: 'antony@gmail.com' } },
    result: {
        data: {
            requestReset: { message: "success", __typename: 'Message' }
        }
    }
}];

describe("<RequestReset/>", () => {
    it("renders and match snapshot", () => {
        const wrapper = mount(
            <MockedProvider>
                <RequestReset />
            </MockedProvider>
        );
        expect(toJSON(wrapper.find("form"))).toMatchSnapshot();
    });

    it("calls the mutation", async () => {
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <RequestReset />
            </MockedProvider>
        );
        wrapper.find('input').simulate('change', { target: { name: 'email', value: 'antony@gmail.com' } });
        wrapper.find("form").simulate('submit');
        await wait();
        wrapper.update();
        expect(toJSON(wrapper.find('p'))).toMatchSnapshot();
    });
});