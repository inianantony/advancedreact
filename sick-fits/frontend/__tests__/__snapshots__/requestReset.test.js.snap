// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> calls the mutation 1`] = `
<p>
  Success! Check your email for reset link
</p>
`;

exports[`<RequestReset/> renders and match snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  method="POST"
  onSubmit={[Function]}
>
  <DisplayError
    error={Object {}}
  />
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign In
    </h2>
    <label
      htmlFor="email"
    >
      Email
      <input
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </label>
    <button
      type="Submit"
    >
      Reset Password
    </button>
  </fieldset>
</form>
`;
